spec/enumerable_spec.rb

```diff
@@ -0,0 +1,160 @@
+      array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(array.my_each_with_index(999)).to eql(nil)
+    end
+
+    it "should return from the end of the array with backward when the index value is negative" do
+      array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      counter = array.length - 1
+      backward_counter = -1
+      array.length.times do
+        expect(array.my_each_with_index(backward_counter)).to eql(array[counter])
+        counter -= 1
+        backward_counter -= 1
+      end
+    end
+  end
+
+  describe "#my_select" do
+    it "should return a new list that contains elements that matches with the given block condition" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      even_numbers = numbers.my_select { |e| e.even? }
+      expect(even_numbers).to be_an_instance_of(Array)
+      expect(even_numbers).to eql([2, 4, 6, 8, 10])
+    end
+  end
+
+  describe "#my_all?" do
+    it "should return a boolean" do
+      expect([1, 2, 3].my_all? { }).to be(true).or be(false)
+    end
+
+    it "should return true if each element matches with given condition" do
+      positive_numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(positive_numbers.my_all? { |e| e > 0 }).to eql(true)
+    end
+
+    it "should return false if each element doesn't match with the given block condition" do
+      numbers = [-3, -2, -1, 0, 1, 2, 3]
+      expect(numbers.my_all? { |e| e > 0 }).to eql(false)
+    end
+  end
+
+
+  describe "#my_any?" do
+    it "Should return a boolean" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(numbers.my_any? { |e| e.even? }).to be(true).or be(false)
+    end
+
+    it "should return true if any element provides the given block condition" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, -10]
+      expect(numbers.my_any? { |e| e < 0 }).to eql(true)
+    end
+
+    it "should return false if any element doesn't provide the given block condition" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(numbers.my_any? { |e| e < 0 }).to eql(false)
+    end
+  end
+
+  describe "#my_none?" do
+    it "should return a boolean" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(numbers.my_none? { |e| e.even? }).to be(true).or be(false)
+    end
+
+    it "should return true if any element doesn't provide the given block condition" do
+      numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(numbers.my_none? { |e| e < 0 }).to eql(true)
+    end
+
+    it "should return false if any element provides the given block condition" do
+      numbers = [-3, -2, -1, 0, 1, 2, 3]
+      expect(numbers.my_none? { |e| e < 0 }).to eql(false)
+    end
+  end
+
+  describe "#my_count" do
+    it "should return 0 when array is empty" do
+      empty_array = []
+      expect(empty_array.my_count).to eql(0)
+    end
+
+    it "should return size of array" do
+      array = [1,2,3,4,5]
+      expect(array.my_count).to eql(array.length)
+      expect(array.my_count).to eql(5)
```

Comment:
I think you are fine checking just against array.length

Reply:
Changed it